radix-engine/src/model/resource_def.rs

```diff
@@ -8,6 +8,16 @@ use scrypto::rust::vec::Vec;
 
 use crate::model::{Proof, ResourceAmount};
 
+pub enum ResourceControllerMethod {
+    Mint,
+    Burn,
+    TakeFromVault,
+    UpdateFlags,
+    UpdateMutableFlags,
```

Comment:
Just a note: Beyond the scope of this PR, I found `UpdateFlags` and `UpdateMutableFlags` confusing this time (although I made the first iteration of implementation). Ideally, I would like `EnableFlags`, `DisableFlags` and `FinalizeFlags`.